### PR TITLE
Add routes file endpoint

### PR DESCRIPTION
**Pull request by edilson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -247,94 +251,94 @@ See Strava API docs for returned data structures.
...
- * `strava.runningRaces.listRaces(args,done)`
+- `strava.runningRaces.get(args,done)`
+- `strava.runningRaces.listRaces(args,done)`
 
 #### Routes
 
-
- * `strava.routes.get(args,done)`
+- `strava.routes.get(args,done)`
+- `strava.routes.getFile(args, done)`
```
Comment (markstos) [inline README.md:319]: Although the documentation for existing methods is often poor, let's do better with new routes, by showing here that the args for `getFile()` are `id` and `file_type`.

File: lib/routes.js [hunk 2]
```diff
@@ -9,14 +9,34 @@ routes.prototype.get = function (args, done) {
...
+
+  _requireRouteId(args)
+
+  this._getFileHelper(endpoint, args, done)
+}
 //= ==== routes endpoint =====
 
+//= ==== helpers =====
+var _requireRouteId = function (args) {
+  if (typeof args.id === 'undefined') {
```
Comment (markstos) [inline lib/routes.js:29]: I believe the ID is always a number. Please strengthen the argument check here to check that it's a defined and a number.

**Review by markstos — CHANGES_REQUESTED** [review]
Looks good. I have requests for some minor updates. Also, please rebase to squash the "merge conflict solution" commit to the other commit.

For future contributions, please submit formatting/whitespace commits separately from logic changes so both types of commits are easier to review.

Thanks.

File: lib/routes.js (hunk 2 above)
Comment (edilson) [reply lib/routes.js:29]: It is actually a long. To test I had to pass it as a string.
https://developers.strava.com/docs/reference/#api-Routes-getRouteAsGPX

**Comment by markstos** [comment]
Please rebase and squash to address the conflicts and this is ready to merge. Thanks for the contribution!

**Comment by edilson** [comment]
@markstos thanks, I will. Haven't done it yet because something strange happened with mu rebase.

**Comment by edilson** [comment]
@markstos you think this is ready to merge now?

File: README.md (hunk 1 above)
Comment (markstos) [reply README.md]: Also, please be consistent abut whether there is a space ",done" or ", done". Looks like the rest of the file doesn't use a space.

File: README.md [hunk 3]
```diff
@@ -247,94 +251,94 @@ See Strava API docs for returned data structures.
...
- * `strava.segments.get(args,done)`
- * `strava.segments.listStarred(args,done)`
- * `strava.segments.listEfforts(args,done)`
- * `strava.segments.listLeaderboard(args,done)`
- * `strava.segments.explore(args,done)` *Expects arg `bounds` as a comma separated string, for two points describing a rectangular boundary for the search: `"southwest corner latitutde, southwest corner longitude, northeast corner latitude, northeast corner longitude"`*.
+- `strava.segments.get(args,done)`
+- `strava.segments.listStarred(args,done)`
+- `strava.segments.listEfforts(args,done)`
+- `strava.segments.listLeaderboard(args,done)`
+- `strava.segments.explore(args,done)` _Expects arg `bounds` as a comma separated string, for two points describing a rectangular boundary for the search: `"southwest corner latitutde, southwest corner longitude, northeast corner latitude, northeast corner longitude"`_.
```
Comment (markstos) [inline README.md:328]: Both "*" and "_" create valid italics in Markdown. Why change it here?

Imagine if each time a feature was merged, the entire README also got reformatted to switch from one valid Markdown syntax to another.

Please remove all the Markdown reformatting from this PR. If you feel strongly our Markdown would better with these apparently arbitrary changes, submit a separate PR for that.

**Review by markstos — CHANGES_REQUESTED** [review]
See commit commits for requested changes.

**Comment by markstos** [comment]
Thanks for the contributions. Published just now in strava-v3@2.1.0.
